docs/source/working_with_aiida/caching.rst

```diff
@@ -65,88 +113,69 @@ This means that caching is enabled for ``TemplatereplacerCalculation`` and ``Str
-2. Get all nodes which match your node, and clear their hash:
+1. Check that the node in question has a ``_aiida_cached_from`` extra. 
+   If that's not the case, it anyhow is not reused.
+2. Clear the hashes of all nodes that are considered identical to your node:
 
     .. code:: python
 
         for n in node.get_all_same_nodes():
             n.clear_hash()
-3. Run your calculation again. Now it should not use caching.
-
-If you instead think that there is a bug in the AiiDA implementation, please open an issue (with enough information to be able to reproduce the error, otherwise it is hard for us to help you) in the AiiDA GitHub repository: https://github.com/aiidateam/aiida_core/issues/new.
-
-.. _caching_provenance:
-
-Caching and the Provenance Graph
---------------------------------
-
-The goal of the caching mechanism is to speed up AiiDA calculations by re-using duplicate calculations. However, the resulting provenance graph should be exactly the same as if caching was disabled. This has important consequences on the kind of caching operations that are possible.
-
-The provenance graph consists of nodes describing data, calculations and workchains, and links describing the relationship between these nodes. We have seen that the hash of a node is used to determine whether two nodes are equivalent. To successfully use a cached node however, we also need to know how the new node should be linked to its parents and children.
-
-In the case of a plain data node, this is simple: Copying a data node from an equivalent node should not change its links, so we just need to preserve the links which this new node already has.
-
-For calculations, the situation is a bit more complex: The node can have inputs and creates new data nodes as outputs. Again, the new node needs to keep its existing links. For the outputs, the calculation needs to create a copy of each node and link these as its outputs. This makes it look as if the calculation had produced these outputs itself, without caching.
+3. Run your calculation again. The node in questions should no longer be reused.
 
-Finally, workchains can create links not only to nodes which they create themselves, but also to nodes created by a calculation that they called, or even their ancestors. This is where caching becomes impossible. Consider the following example (using workfunctions for simplicity):
 
-.. code:: python
 
-    from aiida.engine import workfunction
-    from aiida.orm import Int
+.. _caching_limitations:
 
-    @workfunction
-    def select(a, b):
-        return b
+Limitations
+-----------
 
-    d = Int(1)
-    r1 = select(d, d)
-    r2 = select(Int(1), Int(1))
+1. The current implementation of caching for data nodes clones not only the graph representation of the reused node but also the underlying data in the database and file repository. 
```

Comment:
I find this still more confusing than that it helps. Even when we implement the automated deduplication in the future, what will then be the definition of a "cached" `Data` node. It will still simply be a clone that happens to have the same hash as some other node. There is no link between them whatsoever. I really think that this stems from the fact that we are using the terms "hashing" and "caching" too loosely and interchangeably as explained in one of the first comments in this review. I would simply remove this entire point about `Data` nodes

Reply:
After we had our chat, I now understand what you mean.
I think it is still important to mention here but I've adapted the terminology.